drivers/gpio/gpio_mcux_igpio.c

```diff
@@ -18,57 +18,51 @@ struct mcux_igpio_config {
 };
 
 struct mcux_igpio_data {
+	/* gpio_driver_data needs to be first */
+	struct gpio_driver_data general;
 	/* port ISR callback routine address */
 	sys_slist_t callbacks;
 	/* pin callback routine enable flags, by pin number */
 	u32_t pin_callback_enables;
 };
 
+static int mcux_igpio_pin_interrupt_configure(struct device *dev,
+		unsigned int pin, enum gpio_int_mode mode,
+		enum gpio_int_trig trig);
+
 static int mcux_igpio_configure(struct device *dev,
 			       int access_op, u32_t pin, int flags)
 {
 	const struct mcux_igpio_config *config = dev->config->config_info;
-	gpio_pin_config_t pin_config;
-	u32_t i;
+	GPIO_Type *base = config->base;
+	u8_t from_pin, to_pin, curr_pin;
 
-	/* Check for an invalid pin configuration */
-	if ((flags & GPIO_INT) && (flags & GPIO_DIR_OUT)) {
-		return -EINVAL;
+	if (((flags & GPIO_INPUT) != 0) && ((flags & GPIO_OUTPUT) != 0)) {
+		return -ENOTSUP;
 	}
 
-	pin_config.direction = ((flags & GPIO_DIR_MASK) == GPIO_DIR_IN)
-		? kGPIO_DigitalInput : kGPIO_DigitalOutput;
-
-	pin_config.outputLogic = 0;
-
-	if (flags & GPIO_INT) {
-		if (flags & GPIO_INT_EDGE) {
-			if (flags & GPIO_INT_ACTIVE_HIGH) {
-				pin_config.interruptMode = kGPIO_IntRisingEdge;
-			} else if (flags & GPIO_INT_DOUBLE_EDGE) {
-				pin_config.interruptMode =
-					kGPIO_IntRisingOrFallingEdge;
-			} else {
-				pin_config.interruptMode =
-					kGPIO_IntFallingEdge;
-			}
-		} else { /* GPIO_INT_LEVEL */
-			if (flags & GPIO_INT_ACTIVE_HIGH) {
-				pin_config.interruptMode = kGPIO_IntHighLevel;
-			} else {
-				pin_config.interruptMode = kGPIO_IntLowLevel;
-			}
-		}
+	if ((flags & GPIO_SINGLE_ENDED) != 0) {
+		return -ENOTSUP;
+	}
+
```

Comment:
It would be nice if the configure function returned `-ENOTSUP` for `GPIO_PULL_UP`, `GPIO_PULL_DOWN` flags. The feature is currently not verified by the testcase but we should soon add the check.